libpod/runtime.go

```diff
@@ -1469,6 +1469,24 @@ func (r *Runtime) GetOCIRuntimePath() string {
 // TODO Once runc has support for cgroups, this function should be removed.
 func cgroupV2Check(configPath string, tmpConfig *RuntimeConfig) error {
 	if !tmpConfig.CgroupCheck && rootless.IsRootless() {
+		if tmpConfig.CgroupManager == SystemdCgroupsManager {
+			// If we are running rootless and the systemd manager is requested, be sure that dbus is accessible
+			session := os.Getenv("DBUS_SESSION_BUS_ADDRESS")
+			hasSession := session != ""
+			if hasSession && strings.HasPrefix(session, "unix:path=") {
+				_, err := os.Stat(strings.TrimPrefix(session, "unix:path="))
+				hasSession = err == nil
+			}
+
+			if !hasSession {
+				logrus.Warningf("The cgroups manager is set to systemd but there is no systemd user session available")
+				logrus.Warningf("You may need to enable lingering with: `loginctl enable-linger %d` (possibily as root)", rootless.GetRootlessUID())
```

Comment:
typo: should be `possibly`

Reply:
What does loginctl enable-linger have to do with this situation.  Isn't this more about a dbus user session was never started so users can not dbus with systemd?

Reply:
These needs to be logrus.Errorf, or else users will not see them, by default.

I would add something like

Note: Certain cgroup management will not work correctly unless you are using systemd as the cgroup-manager.

Reply:
Good point about `Errorf`. 

The `linger` thing is actually the initial reason for this PR; it has to do with `podman run -d` and the podman daemon process and someone running `podman run -d` and then logging out (without `linger`, all user jobs will be terminated). podman tries to run the magic `loginctl` command but it won't work if `polkit` isn't installed. That was the original justification for this PR; the limits thing is apparently another newer reason.

Reply:
I understand why linger would be useful, but I don't see where the issue of having no session bus has anything to do with this issue.  Telling the user to use policykit or at least log into a full user session would be more useful then talking about linger.

Reply:
the issue is about not having a session bus and not being able to contact the systemd user instance.

I've changed the error message now and it also suggests to create a proper user session.

Since we don't automatically set the lingering mode now, if an user with UID != 0 wants to use systemd, there is need to do one of these:

- login and create the user session
- If PolicyKit is available: loginctl enable-linger $UID
- If PolicyKit is not available: sudo loginctl enable-linger $UID